test/Microsoft.AspNet.Mvc.Xml.Test/XmlDataContractSerializerInputFormatterTest.cs

```diff
@@ -296,7 +296,7 @@ public void SetMaxDepth_ThrowsWhenMaxDepthIsBelowOne()
                                                                 typeof(XmlException);
             var expectedMessage = TestPlatformHelper.IsMono ?
                 "Expected element 'TestLevelTwo' in namespace '', but found Element node 'DummyClass' in namespace ''" :
-                "The encoding in the declaration 'UTF-8' does not match the encoding of the document 'utf-16LE'.";
+                "The expected encoding 'utf-8' does not match the actual encoding 'utf-16LE'.";
```

Comment:
Can you add a test case showing that the encoding in `charset` overrides the one in `<?xml version=\"1.0\" encoding=\"UTF-8\"?>`?